packages/cli/src/UserManagement/UserManagementHelper.ts

```diff
@@ -46,3 +79,65 @@ export function sanitizeUser(user: User): PublicUser {
 	const { password, resetPasswordToken, createdAt, updatedAt, ...sanitizedUser } = user;
 	return sanitizedUser;
 }
+
+async function getCredentialIdByName(name: string): Promise<string | number | null> {
+	const credential = await Db.collections.Credentials!.findOne({
+		where: {
+			name,
+		},
+	});
+	if (credential) {
+		return credential.id;
+	}
+	return null;
+}
+
+export async function getUserById(userId: string): Promise<User> {
+	const user = await Db.collections.User!.findOneOrFail({
+		where: { id: userId },
+		relations: ['globalRole'],
+	});
+	return user;
+}
+
+export async function checkPermissionsForExecution(
+	workflow: Workflow,
+	userId: string,
+): Promise<void> {
+	const credentialIds = new Set();
+	const nodeNames = Object.keys(workflow.nodes);
+	nodeNames.forEach((nodeName) => {
+		const node = workflow.nodes[nodeName];
+		if (node.credentials) {
+			const credentialNames = Object.keys(node.credentials);
+			credentialNames.forEach((credentialName) => {
+				const credentialDetail = node.credentials![credentialName];
+				if (!credentialDetail.id) {
+					throw new Error(
+						'Error initializing workflow: credential ID not present. Please open the workflow and save it to fix this error.',
+					);
+				}
+				credentialIds.add(credentialDetail.id.toString());
+			});
+		}
+	});
+
+	// We converted all IDs to string so that the set cannot contain duplicates.
+
+	const ids = Array.from(credentialIds);
+
+	if (ids.length === 0) {
+		return;
+	}
+
+	const credentialCount = await Db.collections.SharedCredentials!.count({
+		where: {
+			user: { id: userId },
+			credentials: In(ids),
+		},
+	});
+
+	if (ids.length !== credentialCount) {
+		throw new Error('One or more of the required credentials was not found in the database.');
+	}
+}
```

Comment:
1. let's add some more in code comments what we are doing here and why (for example the length)

2. let's return a `Promise<boolean>` as well so the function is not just a side-effect 👍🏼